CppCoreGuidelines.md

```diff
@@ -11623,14 +11624,14 @@ Casts are widely (mis) used. Modern C++ has constructs that eliminate the need f
 
 * Use templates
 * Use `std::variant`
-
+* Rely on the well defined, safe, implicit conversions between pointer types
 
 ##### Enforcement
 
 * Force the elimination of C-style casts
-* Warn against named casts
 * Warn if there are many functional style casts (there is an obvious problem in quantifying 'many').
 * The [type profile](#Pro-type-reinterpretcast) bans `reinterpret_cast`.
+* Warn against [unnecessary pointer casts](#Pro-type-unnecessarycast).
```

Comment:
Agreed.  I'll also add an enforcement bullet that a warning should be issued for a pointer cast that could be implicit as a broader version of the CppCoreChecker's warning against using a reinterpret_cast<void*>